geoh5py/objects/surveys/electromagnetics/airborne_tem.py

```diff
@@ -0,0 +1,415 @@
+
+        if isinstance(waveform, np.ndarray):
+            if waveform.ndim != 2 or waveform.shape[1] != 2:
+                raise ValueError(
+                    "Input waveform must be a numpy.ndarray of shape (*, 2)."
+                )
+
+            if self.timing_mark is not None:
+                value = self.metadata["EM Dataset"]["Waveform"]
+            else:
+                value = {}
+
+            value["Discretization"] = [
+                {"current": row[1], "time": row[0]} for row in waveform
+            ]
+
+        else:
+            value = waveform
+
+        self.edit_metadata({"Waveform": value})
+
+    @property
+    def yaw(self) -> float | uuid.UUID | None:
+        """
+        Numeric value or property UUID for the yaw angle of the transmitter loop.
+        """
+        return self.fetch_metadata("yaw")
+
+    @yaw.setter
+    def yaw(self, value: float | uuid.UUID):
+        self.set_metadata("yaw", value)
+
+
+class AirborneTEMReceivers(BaseAirborneTEM):
+    """
+    Airborne time-domain electromagnetic receivers class.
+    """
+
+    __TYPE_UID = uuid.UUID("{19730589-fd28-4649-9de0-ad47249d9aba}")
+
+    def __init__(self, object_type: ObjectType, **kwargs):
+        super().__init__(object_type, **kwargs)
+
+    def copy(self, parent=None, copy_children: bool = True) -> AirborneTEMReceivers:
+        """
+        Function to copy a AirborneTEMReceivers to a different parent entity.
+
+        :param parent: Target parent to copy the entity under. Copied to current
+            :obj:`~geoh5py.shared.entity.Entity.parent` if None.
+        :param copy_children: Create copies of AirborneTEMReceivers along with it.
+
+        :return entity: Registered AirborneTEMReceivers to the workspace.
+        """
+        if parent is None:
+            parent = self.parent
+
+        omit_list = ["_metadata", "_receivers", "_transmitters"]
+        new_entity = parent.workspace.copy_to_parent(
+            self, parent, copy_children=copy_children, omit_list=omit_list
+        )
+        new_transmitters = parent.workspace.copy_to_parent(
+            self.transmitters,
+            parent,
+            copy_children=copy_children,
+            omit_list=omit_list,
+        )
+        new_entity.transmitters = new_transmitters
+        parent.workspace.finalize()
+
+        return new_entity
+
+    @classmethod
+    def default_type_uid(cls) -> uuid.UUID:
+        """
+        :return: Default unique identifier
+        """
+        return cls.__TYPE_UID
+
+    @property
+    def default_transmitter_type(self):
+        """
+        :return: Transmitter class
+        """
+        return AirborneTEMTransmitters
+
+    @property
+    def receivers(self) -> AirborneTEMReceivers:
+        """
+        The associated TEM receivers.
+        """
+        return self
+
+    @receivers.setter
+    def receivers(self, value: BaseEMSurvey):
+        raise UserWarning(
```

Comment:
Are you trying to raise an exception here or warn? I think the raise here would interrupt the program. If you want to issue a warning and continue, use the warning module. warnings.warn("Some message.") will issue a warning with the default return object being a UserWarning and continue executing the rest of the code.